CONTRIBUTING.md

```diff
@@ -29,3 +29,138 @@ Recommendations and requirements for how to best contribute to Parquet. We striv
+   compatibility (old readers can still read files).
+
+2. New encodings should be fully specified in this repository and ideally not
+   rely on an external dependencies for implementation (i.e. `parquet-format` is
+   the source of truth for the encoding).
+
+3. New compression mechanisms must have a pure Java implementation that can be
+   used as dependency in parquet-java.
+
+### Releases
+
+The Parquet PMC aims to do releases of the format package only as needed when
+new features are introduced. If multiple new features are being proposed
+simultaneously some features might be consolidated into the same release.
+Guidance is provided below on when implementations should enable features added
+to the specification.  Due to confusion in the past over parquet versioning it
+is not expected that there will be a 3.x release of the specification in the
+foreseeable future.
+
+### Compatibility and Feature Enablement
+
+For the purposes of this discussion we classify features into the following buckets:
+
+1. Backwards compatible. A file written under an older version of the format
+   should be readable under a newer version of the format.
+
+2. Forwards compatible. A file written under a newer version of the format with
+   the enabled feature can be read under an older version of the format, but
+   some information might be missing or performance might be suboptimal.
+
+3. Forward incompatible. A file written under a new version of the format with
+   the feature enabled cannot be read under an older version of the format (e.g.
+   Adding a new compression algorithm).
+
+New features are intended to be widely beneficial to users of Parquet, and
+therefore it is hoped third-party implementations will adopt them quickly after
+they are introduced. It is assumed that writing new parts of the format, and
+especially forward incompatible features, will be configured with feature flag
+defaulted to "off" and at some future point the features are turned on by default
+(reading of the new feature will typically be enabled without configuration or
+defaulted to on). Some amount of lead time is desirable to ensure a critical
+mass of Parquet implementations support a feature to avoid compatability issues
+across the ecosystem.  Therefore, the Parquet PMC gives the following
+recommendations for managing features:
+
+1. Backwards compatibility is the concern of implementations but given the
+   ubiquity of Parquet and the length of time it has been used, libraries should
+   support reading older version of the format to the greatest extent possible.
+
+2. Forward compatible features/changes may be used by default in implementations
+   once the parquet-format containing those changes has been formally released.
+   For features that may pose a significant performance regression to older
+   format readers, libaries should consider delaying default enablement until 1
+   year after the release of the parquet-java implementation that contains the
```

Comment:
I'm still skeptical that this should suggest a precise delay for third-party implementations. Or, alternatively, replace "should" with "may"?

Reply:
I think this is a point of discussion.  I'm OK changing the language either way but would like to get more opinions before doing so.  I think specifying some timeline is important, even if non-binding for third party implementations as it gives third-parties a sense of timeline for adopting features.

Reply:
Perhaps we can either invite more parties to this discussion, or raise it on the ML?

Reply:
Yes, I'll plan on doing one more pass to address open comments then start a discuss thread on the mailing list (hopefully I will have time to do this today)